### PR TITLE
Limit image size in `CommentMarkdownContainer`

### LINKED ISSUE #24569: Image not scaled down in beatmap comment section
### Type

Cosmetic

### Bug description

Comment in osu lazer browser

Qlufwkr
Comment on the website

nXMaJy1

Comment section: **https://osu.ppy.sh/beatmapsets/1512813#osu/3103322**

### Screenshots or videos

_No response_

### Version

osu! 2023.815.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/12359197/database.log)
[input.log](https://github.com/ppy/osu/files/12359198/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/12359199/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/12359200/network.log)
[performance.log](https://github.com/ppy/osu/files/12359201/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12359202/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12359203/updater.log)

### PR DESCRIPTION
**Pull request by EVAST9919**
Closes https://github.com/ppy/osu/issues/24569
Снимок экрана (51)

Stumbled upon some issues along the way, however workarounds are not as bad as I thought they would be.
1. **`Sprite` isn't having a way to limit it's size, unless it's relatively-sized or strictly-defined.**
Fine, we'll make a parent container to do the job then.
2. **Parent container is autosized.**
Tha's okay, we'll set `AutoSizeAxes` to `None` and set it's size manually depending on the provided texture.
3. **The size of DelayedLoadWrapper depends on AutoSizeAxes of it's content.**
This is a bad one and probably needs to be adressed [framework-side](https://github.com/ppy/osu-framework/blob/90d9286f8b209ef9a58620f89875c258702d1732/osu.Framework/Graphics/Containers/DelayedLoadWrapper.cs#L80).
Workaround was simple, we just set it's size after content being loaded.

### CODE REVIEW

File: osu.Game/Overlays/Comments/CommentMarkdownContainer.cs [hunk 1]
```diff
@@ -1,9 +1,17 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+#nullable disable
```
Comment (bdach) [inline osu.Game/Overlays/Comments/CommentMarkdownContainer.cs:4]: Why? Please remove and apply `!` liberally as needed, or at least only apply locally to the nested class.

File: osu.Game/Overlays/Comments/CommentMarkdownContainer.cs [hunk 2]
```diff
@@ -40,5 +50,64 @@ protected override float GetFontSizeByLevel(int level)
...
+                        wrapper.Size = container.Size;
+                        d.FadeInFromZero(300, Easing.OutQuint);
+                    };
+                    return container;
+                }
+
+                private partial class CommentImageContainer : ImageContainer
+                {
+                    // https://github.com/ppy/osu-web/blob/3bd0f406dc78d60b356d955cd4201f8c3e1cca09/resources/css/bem/osu-md.less#L36
+                    // Web version defines max height in em units (6 em), which assuming default pixel size as 14 results in 84 px,
```
Comment (bdach) [inline osu.Game/Overlays/Comments/CommentMarkdownContainer.cs:85]: What is "pixel size" referring to here? `em` units are font-size-dependent.

File: osu.Game/Overlays/Comments/CommentMarkdownContainer.cs [hunk 3]
```diff
@@ -40,5 +50,64 @@ protected override float GetFontSizeByLevel(int level)
...
+                    // which also seems to match my observations upon expecting the web element.
```
Comment (bdach) [inline osu.Game/Overlays/Comments/CommentMarkdownContainer.cs:86]:
```suggestion
                    // which also seems to match observations upon inspecting the web element.
```

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
> 3\. **The size of DelayedLoadWrapper depends on AutoSizeAxes of it's content.**
>         This is a bad one and probably needs to be adressed [framework-side](https://github.com/ppy/osu-framework/blob/90d9286f8b209ef9a58620f89875c258702d1732/osu.Framework/Graphics/Containers/DelayedLoadWrapper.cs#L80).

Why is this "bad"? How else could auto-sized content inside DLW be supported? What would be your proposed replacement?

**Comment by EVAST9919** [comment]
> Why is this "bad"? How else could auto-sized content inside DLW be supported? What would be your proposed replacement?

There's nothing wrong with auto-sized content, I was talking about the case such as this one, in which we need to define the wrapper size manually because content `AutoSizeAxes = None` and so is the wrapper's, because otherwise wrapper size would be 0.
May be allow manually setting AutoSizeAxes, may be copy the size of the content or may be leave it as-is - idk, I'm just pointing this out.

**Review by bdach — APPROVED** [review]

**Pull request merged**